src/cli.rs

```diff
@@ -813,15 +827,19 @@ impl Opts {
                 anyhow::Result::<_, anyhow::Error>::Ok((name, diff))
             });
 
+        let mut has_diff = false;
         stream::iter(update_iter)
             .buffer_unordered(opts.max_concurrent_downloads)
             .try_filter(|(_, diff)| future::ready(diff.is_empty().not()))
             .try_collect::<Vec<_>>()
             .await?
             .into_iter()
-            .for_each(|(name, diff)| print_diff(name, diff));
+            .for_each(|(name, diff)| {
+                has_diff |= !diff.is_empty();
+                print_diff(name, diff);
+            });
 
-        if !opts.dry_run {
+        if has_diff && !opts.dry_run {
```

Comment:
Since the empty differences are already filtered out you could do it like this instead:
```rust
        let differences = stream::iter(update_iter)
            .buffer_unordered(opts.max_concurrent_downloads)
            .try_filter(|(_, diff)| future::ready(diff.is_empty().not()))
            .try_collect::<Vec<_>>()
            .await?;

        for (name, diff) in &differences {
            print_diff(name, diff);
        }

        if !differences.is_empty() && !opts.dry_run {
```